internals-js/src/federation.ts

```diff
@@ -487,6 +510,10 @@ export class FederationMetadata {
     return this.sharingPredicate()(field);
   }
 
+  isKeyField(field: FieldDefinition<any>): boolean {
```

Comment:
Following my other comment on `merge.ts`, I don't think this method is currently necessary (since it's a strict subset of `isUsedField`). So maybe we can remove it and `computeKeys` for now to avoid dead code?